quil-rs/src/expression/mod.rs

```diff
@@ -106,50 +108,52 @@ impl PrefixExpression {
     }
 }
 
+impl PartialEq for Expression {
+    // Implemented by hand since we can't derive with f64s hidden inside.
+    fn eq(&self, other: &Self) -> bool {
+        match (self, other) {
+            (Self::Address(left), Self::Address(right)) => left == right,
+            (Self::Infix(left), Self::Infix(right)) => left == right,
+            (Self::Number(left), Self::Number(right)) => left == right,
+            (Self::Prefix(left), Self::Prefix(right)) => left == right,
+            (Self::FunctionCall(left), Self::FunctionCall(right)) => left == right,
+            (Self::Variable(left), Self::Variable(right)) => left == right,
+            (Self::PiConstant, Self::PiConstant) => true,
+            _ => false,
+        }
+    }
+}
+
+// Implemented by hand since we can't derive with f64s hidden inside.
+impl Eq for Expression {}
+
 impl Hash for Expression {
     // Implemented by hand since we can't derive with f64s hidden inside.
-    // Also to understand when things should be the same, like with commutativity (`1 + 2 == 2 + 1`).
-    // See https://github.com/rigetti/quil-rust/issues/27
     fn hash<H: Hasher>(&self, state: &mut H) {
-        use std::cmp::{max_by_key, min_by_key};
-        use Expression::*;
         match self {
-            Address(m) => {
+            Expression::Address(m) => {
```

Comment:
Small style nit, but while we're here: should use `Self` instead of `Expression`.

(It's fine not to make this change if there isn't anything more substantial being made. It's just a style thing that is triggered by a certain clippy lint)

Reply:
Thanks! 

Yikes, I just ran `cargo clippy -- -W clippy::pedantic` 😮